### PR TITLE
[integration_test/postgres/sles] Modify sles 15 install

### PR DESCRIPTION
**Pull request by schmikei**
## Description
Inserts more data into installed postgres and simplifies install of postgres 14 now that it is more supported on SLES 15.

## Related issue
NA

## How has this been tested?
Kokoro testing.

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: integration_test/third_party_apps_data/applications/postgresql/sles/install [hunk 1]
```diff
@@ -5,31 +5,19 @@ SUSE_VERSION="${VERSION_ID%%.*}"
...
-  sudo /usr/pgsql-14/bin/postgresql-14-setup initdb
-  sudo service postgresql-14 restart
+sudo su postgres -c "psql postgres -c \"ALTER ROLE postgres WITH PASSWORD 'abc123';\""
 
-  sudo su postgres -c "psql postgres -c \"ALTER ROLE postgres WITH PASSWORD 'abc123';\""
-fi
+sudo sh -c 'echo "host    all             all             127.0.0.1/32            scram-sha-256" > /var/lib/pgsql/data/pg_hba.conf'
+sudo sh -c 'echo "host    all             all             ::1/128                 scram-sha-256" >> /var/lib/pgsql/data/pg_hba.conf'
 
+sudo service postgresql restart
```
Comment (igorpeshansky) [inline integration_test/third_party_apps_data/applications/postgresql/sles/install:23]: Do we need the second `restart` to pick up the content of `pg_hba.conf`? Can we instead create that file before the first `restart`?

File: integration_test/third_party_apps_data/applications/postgresql/sles/install [hunk 2]
```diff
@@ -5,31 +5,19 @@ SUSE_VERSION="${VERSION_ID%%.*}"
...
 
-  sudo /usr/pgsql-14/bin/postgresql-14-setup initdb
-  sudo service postgresql-14 restart
+sudo su postgres -c "psql postgres -c \"ALTER ROLE postgres WITH PASSWORD 'abc123';\""
 
-  sudo su postgres -c "psql postgres -c \"ALTER ROLE postgres WITH PASSWORD 'abc123';\""
-fi
+sudo sh -c 'echo "host    all             all             127.0.0.1/32            scram-sha-256" > /var/lib/pgsql/data/pg_hba.conf'
+sudo sh -c 'echo "host    all             all             ::1/128                 scram-sha-256" >> /var/lib/pgsql/data/pg_hba.conf'
```
Comment (igorpeshansky) [inline integration_test/third_party_apps_data/applications/postgresql/sles/install:21]: [Optional] You can avoid the `sh -c` wrapper by using `cat | sudo tee`:
```suggestion
cat <<EOF | sudo tee /var/lib/pgsql/data/pg_hba.conf >/dev/null
host    all             all             127.0.0.1/32            scram-sha-256
host    all             all             ::1/128                 scram-sha-256
EOF
```

**Review by igorpeshansky — APPROVED** [review]
> …this was what you were expecting in https://github.com/GoogleCloudPlatform/ops-agent/pull/804#discussion_r966323115?

Yes, exactly. Thanks.

LGTM :shipit:

File: integration_test/third_party_apps_data/applications/postgresql/sles/install (hunk 1 above)
Comment (schmikei) [reply integration_test/third_party_apps_data/applications/postgresql/sles/install]: Turns out we have to start postgres to generate that file so I added the start before the file insertion

**Comment by schmikei** [comment]
Looking into new failures, may take a second though

**Comment by schmikei** [comment]
Turns out the `cat <<EOF | sudo tee...` command was overwriting all content, reverting back to the echos

**Comment by qingling128** [comment]
Looking at the two failed tests:

### `Ops Agent integration test (Windows)` [Details](https://source.cloud.google.com/results/invocations/c27ccb94-2d64-40cd-bd22-11ca0b059ca0)

Failed for:
```
QueryLog() failed: windows_event_log not found, exhausted retries
```

```
2022/09/09 02:35:41 logName=~"projects/stackdriver-test-143416/logs/windows_event_log" AND resource.labels.instance_id="1634594367684943261" AND timestamp > "2022-09-09T01:35:41Z" AND jsonPayload.Message:application_msg
2022/09/09 02:35:42 Query returned found=false, err=<nil>, attempt=80
```

The error above is unrelated to this PR.

### `third party apps integration test (SLES 12)` [Details](https://source.cloud.google.com/results/invocations/9e078880-1aec-4877-9653-6549e0d449a1)

Failed for:

```
--- FAIL: TestThirdPartyApps (0.18s)
    --- FAIL: TestThirdPartyApps/sles-12/postgresql (985.76s)
```

```
2022/09/09 01:56:28 logName=~"projects/stackdriver-test-143416/logs/postgresql_general" AND resource.labels.instance_id="1957173723311581025" AND timestamp > "2022-09-09T00:56:28Z" AND jsonPayload.message=~".*starting PostgreSQL.*" AND jsonPayload.level=~"LOG" AND labels."logging.googleapis.com/instrumentation_source"=agent.googleapis.com/postgresql_general
2022/09/09 01:56:28 Query returned found=false, err=<nil>, attempt=80
```

The change seems to have fixed SLES 15, but not working on SLES 12. For comparison, "TestThirdPartyApps/sles-12/postgresql" in master passed yesterday (back at commit https://github.com/GoogleCloudPlatform/ops-agent/commit/32d7597b48fcc3c71f630db570c0a07faad502ff)

**Comment by schmikei** [comment]
I think it should be fixed now however I am confused as to why the service start did not generate the log. The old iteration of the script generated an appropriate entry:

```
keith-sles-17:/var/lib/pgsql/data/log # cat /etc/os-release
NAME="SLES"
VERSION="12-SP5"
VERSION_ID="12.5"
PRETTY_NAME="SUSE Linux Enterprise Server 12 SP5"
ID="sles"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:12:sp5"
keith-sles-17:/var/lib/pgsql/data/log # cat postgresql*.log
2022-09-09 08:24:10.885 UTC   [2393]LOG:  starting PostgreSQL 14.1 on x86_64-suse-linux-gnu, compiled by gcc (SUSE Linux) 4.8.5, 64-bit
```

I think what is current is probably good and consistent enough to work though as it also passed in #804

**Comment by qingling128** [comment]
The windows tests failed for:

```
    ops_agent_test.go:1277: setupOpsAgent() failed to restart ops agent: Command failed: Restart-Service google-cloud-ops-agent -Force
        Command failed: [/tmp/tmp.O3aZTEyyal/winrm.par --host=10.128.0.131 --username=windows_user --password=@;YUkq5,h&r9#\K --command=powershell -NonInteractive -encodedcommand "UgBlAHMAdABhAHIAdAAtAFMAZQByAHYAaQBjAGUAIABnAG8AbwBnAGwAZQAtAGMAbABvAHUAZAAtAG8AcABzAC0AYQBnAGUAbgB0ACAALQBGAG8AcgBjAGUA" --stderrthreshold=fatal --verbosity=-2]
        exit status 1
        stdout+stderr: #< CLIXML
        <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Restart-Service : Time out has expired and the operation has not been _x000D__x000A_</S><S S="Error">completed._x000D__x000A_</S><S S="Error">At line:1 char:1_x000D__x000A_</S><S S="Error">+ Restart-Service google-cloud-ops-agent -Force_x000D__x000A_</S><S S="Error">+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : NotSpecified: (:) [Restart-Service], TimeoutExce _x000D__x000A_</S><S S="Error">   ption_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : System.ServiceProcess.TimeoutException,Microsoft _x000D__x000A_</S><S S="Error">   .PowerShell.Commands.RestartServiceCommand_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
```

It's probably unrelated to this PR and related to this other known issue we are investigating b/240564518 (windows service start / restart failure)

**Review by qingling128 — APPROVED** [review]
LGTM

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

File: integration_test/third_party_apps_data/applications/postgresql/sles/install (hunk 2 above)
Comment (igorpeshansky) [reply integration_test/third_party_apps_data/applications/postgresql/sles/install:21]: I wonder why this ended up not working, but not a big deal either way…

**Pull request merged**